### ISSUE TITLE
All output of type 'indicator' contain 'PLACEHOLDER' in 'pattern'

### ISSUE THREAD

**Issue by StephenThomson** [oasis-open/cti-stix-elevator#55]
In my testing it seems that indicators don't elevate correctly and are output containing PLACEHOLDER.

Input:
```
<stix:STIX_Package xmlns:cyboxCommon="http://cybox.mitre.org/common-2" xmlns:cybox="http://cybox.mitre.org/cybox-2" xmlns:cyboxVocabs="http://cybox.mitre.org/default_vocabularies-2" xmlns:marking="http://data-marking.mitre.org/Marking-1" xmlns:simpleMarking="http://data-marking.mitre.org/extensions/MarkingStructure#Simple-1" xmlns:tlpMarking="http://data-marking.mitre.org/extensions/MarkingStructure#TLP-1" xmlns:TOUMarking="http://data-marking.mitre.org/extensions/MarkingStructure#Terms_Of_Use-1" xmlns:opensource="http://hailataxii.com" xmlns:edge="http://soltra.com/" xmlns:indicator="http://stix.mitre.org/Indicator-2" xmlns:ttp="http://stix.mitre.org/TTP-1" xmlns:stixCommon="http://stix.mitre.org/common-1" xmlns:stixVocabs="http://stix.mitre.org/default_vocabularies-1" xmlns:stix="http://stix.mitre.org/stix-1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:taxii="http://taxii.mitre.org/messages/taxii_xml_binding-1" xmlns:taxii_11="http://taxii.mitre.org/messages/taxii_xml_binding-1.1" xmlns:tdq="http://taxii.mitre.org/query/taxii_default_query-1" id="edge:Package-0d6e49eb-35e8-43c7-824d-5aefbb387b40" version="1.1.1" timestamp="2016-02-09T12:35:51.135458+00:00">
    <stix:STIX_Header>
        <stix:Handling>
            <marking:Marking>
                <marking:Controlled_Structure>../../../../descendant-or-self::node()</marking:Controlled_Structure>
                <marking:Marking_Structure xsi:type="tlpMarking:TLPMarkingStructureType" color="WHITE"/>
                <marking:Marking_Structure xsi:type="TOUMarking:TermsOfUseMarkingStructureType">
                    <TOUMarking:Terms_Of_Use>www.malwaredomainlist.com | Malware Domain List - is a non-commercial community project. Our list can be used for free by anyone. Feel free to use it. 
</TOUMarking:Terms_Of_Use>
                </marking:Marking_Structure>
                <marking:Marking_Structure xsi:type="simpleMarking:SimpleMarkingStructureType">
                    <simpleMarking:Statement>Unclassified (Public)</simpleMarking:Statement>
                </marking:Marking_Structure>
            </marking:Marking>
        </stix:Handling>
    </stix:STIX_Header>
    <stix:Indicators>
        <stix:Indicator id="opensource:indicator-4c72329f-dc3a-4582-b872-3e6f469b2260" timestamp="2015-02-05T11:34:30.219739+00:00" xsi:type="indicator:IndicatorType" version="2.1.1">
            <indicator:Title>Compromised Site: : qwmlad.xyz:9290/openvpnadmin/popular.php?shopping=26</indicator:Title>
            <indicator:Type xsi:type="stixVocabs:IndicatorTypeVocab-1.1">IP Watchlist</indicator:Type>
            <indicator:Type xsi:type="stixVocabs:IndicatorTypeVocab-1.1">Domain Watchlist</indicator:Type>
            <indicator:Type xsi:type="stixVocabs:IndicatorTypeVocab-1.1">URL Watchlist</indicator:Type>
            <indicator:Description>This domain qwmlad.xyz:9290 (91.209.77.124) located in RU (Russia), has been identified as Malious [exploit kit] by www.malwaredomainlist.com. For more detailed infomation about this indicator go to [CAUTION!!Read-URL-Before-Click] [http://www.malwaredomainlist.com/mdl.php?search=qwmlad.xyz:9290].</indicator:Description>
            <indicator:Observable idref="opensource:Observable-0bc3ed1b-6ea7-4662-af0c-b6aef817cc42">
            </indicator:Observable>
            <indicator:Indicated_TTP>
                <stixCommon:TTP idref="opensource:ttp-e49fe46a-c86d-4b60-b619-2efef53073a4" xsi:type="ttp:TTPType"/>
            </indicator:Indicated_TTP>
            <indicator:Producer>
                <stixCommon:Identity id="opensource:Identity-f465129c-0151-4f74-8913-b80f6a1516a9">
                    <stixCommon:Name>www.malwaredomainlist.com</stixCommon:Name>
                </stixCommon:Identity>
                <stixCommon:Time>
                    <cyboxCommon:Produced_Time>2015-02-04T13:35:00+00:00</cyboxCommon:Produced_Time>
                    <cyboxCommon:Received_Time>2015-02-05T11:34:30+00:00</cyboxCommon:Received_Time>
                </stixCommon:Time>
            </indicator:Producer>
        </stix:Indicator>
    </stix:Indicators>
</stix:STIX_Package>
```
Output:
`{u'valid_from': u'2015-02-05T11:34:30.219739Z', u'created_by_ref': u'identity--f465129c-0151-4f74-8913-b80f6a1516a9', u'description': u'This domain qwmlad.xyz:9290 (91.209.77.124) located in RU (Russia), has been identified as Malious [exploit kit] by www.malwaredomainlist.com. For more detailed infomation about this indicator go to [CAUTION!!Read-URL-Before-Click] [http://www.malwaredomainlist.com/mdl.php?search=qwmlad.xyz:9290].', u'created': u'2015-02-05T11:34:30.219Z', 'pattern': u'PLACEHOLDER:opensource:Observable-0bc3ed1b-6ea7-4662-af0c-b6aef817cc42', u'labels': [u'ip-watchlist', u'domain-watchlist', u'url-watchlist'], u'modified': u'2015-02-05T11:34:30.219Z', u'object_marking_refs': [u'marking-definition--77ff791e-6cb4-4f03-be04-6c748c453ef3', u'marking-definition--fc1c40e3-9ffd-45b4-8a25-c504898e6414', u'marking-definition--db89b6c0-a0db-4fec-b0d7-d3388b243b27'], u'type': u'indicator', u'id': u'indicator--4c72329f-dc3a-4582-b872-3e6f469b2260', u'name': u'Compromised Site: : qwmlad.xyz:9290/openvpnadmin/popular.php?shopping=26'}`

**Comment by rpiazza** [comment]
Hi Stephen,

Your original STIX 1.1.1 XML file does not contain any information to create a indicator pattern.  Therefore the elevator was unable to convert anything. 

```
<indicator:Observable idref="opensource:Observable-0bc3ed1b-6ea7-4662-af0c-b6aef817cc42">
</indicator:Observable>
```
In general, when the elevator cannot convert an observable, it will mark it in the output as
```
PLACEHOLDER:<id>
```
so it easier to find.

Also, the elevator cannot convert all STIX 1.1.1 content flawlessly because of the differences between STIX 1.x and STIX 2.0.   The elevator outputs log messages to help explain what the issues are.  These two were generated by the elevator when running on the STIX package you included above:

[205] [WARNING] [2017-05-18 12:34:03,765] At least one PLACEHOLDER idref was not resolved in indicator--4c72329f-dc3a-4582-b872-3e6f469b2260
[203] [ERROR  ] [2017-05-18 12:34:03,765] Placeholder opensource:Observable-0bc3ed1b-6ea7-4662-af0c-b6aef817cc42 should be resolved

Perhaps you want to use something like this as the body of that observable:

```
<cybox:Object id="example:object-d7fcce87-0e98-4537-81bf-1e7ca9ad3734">
                <cybox:Properties xsi:type="AddressObject:AddressObjectType" category="ipv4-addr">
                    <AddressObject:Address_Value>91.209.77.124</AddressObject:Address_Value>
                </cybox:Properties>
            </cybox:Object>
```

**Comment by StephenThomson** [comment]
Hi Rich,

Just realised I forgot to thank you at the time for your help with this issue, thank you.

Steve
